complaint_search/es_builders.py

```diff
@@ -41,47 +52,69 @@ def add(self, **kwargs):
     def build(self):
          """Method that will build the body dictionary."""
 
-    def _create_bool_should_clauses(self, es_field_name, value_list, 
-        with_subitems=False, es_subitem_field_name=None):
+    ## This create all the bool should filter clauses for a field
+    ## es_field_name: a field name (to be filtered in ES)
+    ## value_list: a list of values to be filtered for this field
+    ## with_child: set to true if this field has a child field
+    ## es_child_field_name: if with_child is true, this holds the field
+    ## name of the child used (to be filtered in ES)
+    def _build_bool_should_clauses(self, field):
+        es_field_name = self._get_es_name(field)
+        value_list = [ 0 if cd.lower() == "no" else 1 for cd in self.params[field] ] \
+            if field in self._OPTIONAL_FILTERS_STRING_TO_BOOL else self.params.get(field) 
         if value_list:
-            if not with_subitems:
+            if not self._has_child(field):
                 term_list = [ {"terms": {es_field_name: [value]}} 
                     for value in value_list ]
                 return {"bool": {"should": term_list}}
             else:
                 item_dict = defaultdict(list)
                 for v in value_list:
-                    # -*- coding: utf-8 -*-
-                    v_pair = v.split(u'\u2022')
-                    # No subitem
+                    v_pair = v.split(DELIMITER)
+                    # No child specified
                     if len(v_pair) == 1:
-                        # This will initialize empty list for item if not in item_dict yet
+                        # This will initialize empty list for item if not in 
+                        # item_dict yet
                         item_dict[v_pair[0]]
                     elif len(v_pair) == 2:
                         # put subproduct into list
                         item_dict[v_pair[0]].append(v_pair[1])
 
                 # Go through item_dict to create filters
                 f_list = []
-                for item, subitems in item_dict.iteritems():
+                for item, child in item_dict.iteritems():
```

Comment:
I think child works OK here, but we should be careful about this nested filter (parent / child) vs. parent / child field relationships in elasticsearch queries or implying any direct relationship between the two fields in the index as you could technically do this with any two fields as a nested filter.